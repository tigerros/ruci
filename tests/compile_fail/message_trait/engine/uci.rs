fn a(_: impl ruci::engine::traits::Message) {}
fn b(m: ruci::Uci) {
    a(m);
}

fn main() {}