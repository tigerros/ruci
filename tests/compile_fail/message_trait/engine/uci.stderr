error[E0277]: the trait bound `ruci::Uci: ruci::engine::traits::Message` is not satisfied
 --> tests/compile_fail/message_trait/engine/uci.rs:3:7
  |
3 |     a(m);
  |     - ^ the trait `ruci::engine::traits::Message` is not implemented for `ruci::Uci`
  |     |
  |     required by a bound introduced by this call
  |
  = help: the following other types implement trait `ruci::engine::traits::Message`:
            &ruci::BestMove
            &ruci::CopyProtection
            &ruci::Id<'_>
            &ruci::Info<'_>
            &ruci::Option<'_>
            &ruci::ReadyOk
            &ruci::Registration
            &ruci::UciOk
          and $N others
note: required by a bound in `a`
 --> tests/compile_fail/message_trait/engine/uci.rs:1:14
  |
1 | fn a(_: impl ruci::engine::traits::Message) {}
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `a`
